### PR TITLE
acceptance tests: DELETE /devices?device_id

### PR DESCRIPTION
**Pull request by mchalski**
Issues: MEN-1134

some refactoring was required to support both internal and mgmt apis. this is actually most of this PR.

@maciejmrowiec @mendersoftware/rndity

### CODE REVIEW

File: tests/tests/test_delete_device.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from client import ManagementClient, ManagementClientSimple, InternalClient
```
Comment (bboozzoo) [inline tests/tests/test_delete_device.py]: You don't need ManagementClient

File: tests/tests/test_delete_device.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import bravado
```
Comment (bboozzoo) [inline tests/tests/test_delete_device.py]: bravado can go away too

File: tests/tests/test_delete_device.py (hunk 1 above)
Comment (mchalski) [inline tests/tests/test_delete_device.py]: fixed

File: tests/tests/test_delete_device.py (hunk 2 above)
Comment (mchalski) [inline tests/tests/test_delete_device.py]: fixed

**Pull request merged**